### PR TITLE
Vet clinic database: add "join table" for visits

### PR DESCRIPTION
**Pull request by SanaTameem**
# IN this milestone: 
- Create a table named vets with the following columns:
  - id: integer (set it as autoincremented PRIMARY KEY)
  - name: string
  - age: integer
  - date_of_graduation: date

- There is a many-to-many relationship between the tables species and vets: a vet can specialize in multiple species, and a species can have multiple vets specialized in it. Create a "join table" called specializations to handle this relationship.

- There is a many-to-many relationship between the tables animals and vets: an animal can visit multiple vets and one vet can be visited by multiple animals. Create a "join table" called visits to handle this relationship, it should also keep track of the date of the visit.

- Insert the following data for vets:
  - Vet William Tatcher is 45 years old and graduated Apr 23rd, 2000.
  - Vet Maisy Smith is 26 years old and graduated Jan 17th, 2019.
  - Vet Stephanie Mendez is 64 years old and graduated May 4th, 1981.
  - Vet Jack Harkness is 38 years old and graduated Jun 8th, 2008.

- Insert the following data for specializations:
  - Vet William Tatcher is specialized in Pokemon.
  - Vet Stephanie Mendez is specialized in Digimon and Pokemon.
  - Vet Jack Harkness is specialized in Digimon.

- Insert the following data for visits:
  - Agumon visited William Tatcher on May 24th, 2020.
  - Agumon visited Stephanie Mendez on Jul 22th, 2020.
  - Gabumon visited Jack Harkness on Feb 2nd, 2021.
  - Pikachu visited Maisy Smith on Jan 5th, 2020.
  - Pikachu visited Maisy Smith on Mar 8th, 2020.
  - Pikachu visited Maisy Smith on May 14th, 2020.
  - Devimon visited Stephanie Mendez on May 4th, 2021.
  - Charmander visited Jack Harkness on Feb 24th, 2021.
  - Plantmon visited Maisy Smith on Dec 21st, 2019.
  - Plantmon visited William Tatcher on Aug 10th, 2020.
  - Plantmon visited Maisy Smith on Apr 7th, 2021.
  - Squirtle visited Stephanie Mendez on Sep 29th, 2019.
  - Angemon visited Jack Harkness on Oct 3rd, 2020.
  - Angemon visited Jack Harkness on Nov 4th, 2020.
  - Boarmon visited Maisy Smith on Jan 24th, 2019.
  - Boarmon visited Maisy Smith on May 15th, 2019.
  - Boarmon visited Maisy Smith on Feb 27th, 2020.
  - Boarmon visited Maisy Smith on Aug 3rd, 2020.
  - Blossom visited Stephanie Mendez on May 24th, 2020.
  - Blossom visited William Tatcher on Jan 11th, 2021.

- Write queries to answer the following:
  - Who was the last animal seen by William Tatcher?
limit 1

  - How many different animals did Stephanie Mendez see?
query - 2

  - List all vets and their specialties, including vets with no specialties.
query - 3

  - List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
query - 4

  - What animal has the most visits to vets?
query - 5

  - Who was Maisy Smith's first visit?
limit2

  - Details for most recent visit: animal information, vet information, and date of visit.
recent visit

  - How many visits were with a vet that did not specialize in that animal's species?
unspesial

  - What specialty should Maisy Smith consider getting? Look for the species she gets the most.
query - 9

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -88,3 +88,33 @@ SELECT animals.name AS animal_name FROM animals JOIN owners ON animals.owner_id
 
 /*Who owns the most animals?*/
 SELECT owners.full_name, COUNT(animals.id) AS total_animals FROM owners JOIN animals ON owners.id = animals.owner_id GROUP BY owners.full_name;
+
+
+/*******************DAY 4 project*************/
+
+/*Who was the last animal seen by William Tatcher?*/
+SELECT animals.name FROM visits JOIN animals ON visits.animal_id = animals.id WHERE visits.vet_id = 1 ORDER BY visits.visit_date DESC;
```
Comment (franclobo) [inline queries.sql]: - [ ] Kindly use LIMIT 1; to limit the result to the last animal seen by the vet. 👍🏽

File: queries.sql [hunk 2]
```diff
@@ -88,3 +88,33 @@ SELECT animals.name AS animal_name FROM animals JOIN owners ON animals.owner_id
...
+SELECT vets.name, species.name FROM specializations FULL JOIN vets ON specializations.vet_id = vets.id FULL JOIN species ON specializations.species_id = species.id;
+
+/*List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.*/
+SELECT animals.name FROM visits JOIN animals ON visits.animal_id = animals.id WHERE visits.vet_id = 3 AND visits.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
+
+/*What animal has the most visits to vets?*/
+SELECT vets.name, MIN(visits.visit_date) AS first_visit_date FROM vets JOIN visits ON vets.id = visits.vet_id WHERE vets.name = 'Maisy Smith' GROUP BY vets.name;
+
+/*Who was Maisy Smith's first visit?*/
+SELECT animals.name AS animal_name, MIN(visits.visit_date) AS first_visit_date FROM visits JOIN animals ON visits.animal_id = animals.id JOIN vets ON visits.vet_id = vets.id WHERE vets.name = 'Maisy Smith' GROUP BY animals.name ORDER BY first_visit_date;
```
Comment (franclobo) [inline queries.sql]: - [ ] Kindly use LIMIT 1; to limit the result to the first visit to the vet. 👍🏽

File: queries.sql [hunk 3]
```diff
@@ -88,3 +88,33 @@ SELECT animals.name AS animal_name FROM animals JOIN owners ON animals.owner_id
...
+
+/*Details for most recent visit: animal information, vet information, and date of visit.*/
+SELECT animals.name AS animal_name, vets.name AS vet_name, MAX(visits.visit_date) AS most_recent_visit_date FROM visits JOIN animals ON visits.animal_id = animals.id JOIN vets ON visits.vet_id = vets.id GROUP BY animals.name, vets.name;
```
Comment (franclobo) [inline queries.sql]: - [ ] The most recent visit corresponds to Devimon '2021-05-04', please correct this query to have the required information. 👀

File: queries.sql [hunk 4]
```diff
@@ -88,3 +88,33 @@ SELECT animals.name AS animal_name FROM animals JOIN owners ON animals.owner_id
...
+
+/*How many visits were with a vet that did not specialize in that animal's species?*/
+SELECT COUNT(*) FROM visits LEFT JOIN specializations ON visits.animal_id = specializations.species_id AND visits.vet_id = specializations.vet_id WHERE specializations.species_id IS NULL;
```
Comment (franclobo) [inline queries.sql]: - [ ] There are 12 visits with a vet that did not specialize in that animal's species, those are 9 from `Maisy Smith` who doesn't have specialization and another three that don't correspond to the specialization of the vet. Please 🙏🏽 correct this query to have the required information.

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @SanaTameem, 

# Required Changes ♻️

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- Correct schema ✔️ 
- Correct data ✔️ 
- Correct queries ✔️ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by christianbiring1 — APPROVED** [review]
Hi @SanaTameem , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- [ ] Great work on implementing the changes as requested by the previous reviewer :100: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**